superset-frontend/cypress-base/cypress/integration/dashboard/edit_mode.test.js

```diff
@@ -23,47 +23,55 @@ describe('Dashboard edit mode', () => {
     cy.server();
     cy.login();
     cy.visit(WORLD_HEALTH_DASHBOARD);
-    cy.get('.dashboard-header [data-test=edit-alt]').click();
+    cy.get('[data-test="dashboard-header"]').find('[data-test=pencil]').click();
   });
 
   it('remove, and add chart flow', () => {
     // wait for box plot to appear
-    cy.get('.grid-container .box_plot');
+    cy.get('[data-test="grid-container"]').find('.box_plot');
 
-    cy.get('.fa.fa-trash')
+    cy.get('[data-test="icon-button"]')
       .last()
       .then($el => {
         cy.wrap($el).invoke('show').click();
         // box plot should be gone
-        cy.get('.grid-container .box_plot').should('not.exist');
+        cy.get('[data-test="grid-container"]')
+          .find('.box_plot')
+          .should('not.exist');
       });
 
-    cy.get('.tabs-components .nav-tabs li a').contains('Charts').click();
+    cy.get('[data-test="tabs-component"]').children().siblings().last().click();
 
     // wait for tab-switching animation to complete
     cy.wait(1000);
 
     // find box plot is available from list
-    cy.get('.tabs-components')
-      .find('.chart-card-container')
+    cy.get('[data-test="dragdroppable-object"]')
+      .find('[data-test="card-title"]')
```

Comment:
Is the `find()` necessary if the data-test selector is specific enough?

Reply:
it is for better reflecting components structure. In object dragdroppable-object I expect to find card-title. Just a pattern to better see changes in case of some components breakup